### ISSUE TITLE
Composer fails to find php extensions when executed through ST2

### ISSUE THREAD

**Issue by jacksleight** [francodacosta/composer-sublime#2]
When trying to install https://packagist.org/packages/doctrine/mongodb-odm recently I ran into the following error:

```
executing: /bin/composer.phar update -n -v
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - doctrine/mongodb 1.0.x-dev requires ext-mongo >=1.2.12,<1.4-dev -> the requested PHP extension mongo is missing from your system.
    - doctrine/mongodb 1.0.2 requires ext-mongo >=1.2.12,<1.4-dev -> the requested PHP extension mongo is missing from your system.
    - doctrine/mongodb 1.0.1 requires ext-mongo >=1.2.12,<1.4-dev -> the requested PHP extension mongo is missing from your system.
    - doctrine/mongodb 1.0.0-BETA3 requires ext-mongo >=1.3.1,<1.4-dev -> the requested PHP extension mongo is missing from your system.
    - doctrine/mongodb 1.0.0-BETA2 requires ext-mongo >=1.2.12,<1.3-dev -> the requested PHP extension mongo is missing from your system.
    - doctrine/mongodb 1.0.0-BETA1 requires ext-mongo * -> the requested PHP extension mongo is missing from your system.
    - doctrine/mongodb 1.0.0 requires ext-mongo >=1.2.12,<1.4-dev -> the requested PHP extension mongo is missing from your system.
    - Installation request for doctrine/mongodb 1.0.*@dev -> satisfiable by doctrine/mongodb 1.0.0, doctrine/mongodb 1.0.0-BETA1, doctrine/mongodb 1.0.0-BETA2, doctrine/mongodb 1.0.0-BETA3, doctrine/mongodb 1.0.1, doctrine/mongodb 1.0.2, doctrine/mongodb 1.0.x-dev.
```

However `php -m` has:

```
[PHP Modules]
...
mongo
...
```

And `/bin/composer.phar show --platform` has:

```
...
ext-mongo        1.3.5     The mongo PHP extension
...
```

And if I open up Terminal (this is on OS X) and run `/bin/composer.phar update -n -v` it works fine.

My user settings looks like this:

``` javascript
{
    "composer_command":  "/bin/composer.phar"
}
```

**Comment by francodacosta** [comment]
Hi,

I'm afraid I can not help you with that. you are getting an error form composer itself.

it seems it can not satisfy the package dependencies

```
the requested PHP extension mongo is missing from your system.
```

I think you need to manually install the php extension for mongo db

**Comment by jacksleight** [comment]
Hi, 

Sorry I may not have explained it very well. Yes that error does appear, but the php extension for mongo db is already installed correctly, and if I run Composer via Terminal the check for that extension passess and the package is installed correctly.

However, when I try to run composer through the ST2 plugin, for some reason it cannot find the mongo extension, and I see that error. I assume this is something to do with the way that the ST2 plugin is running Composer?

**Comment by francodacosta** [comment]
I did misunderstood you, sorry.

Can you paste your composer.json file, so I can test on my system ?

thanks

nuno

**Comment by jacksleight** [comment]
Sure, this is the bear minimum that produces the error (just composer.json in an otherwise empty directory):

``` javascript
{
    "require": {
        "doctrine/mongodb-odm": "1.0.*@dev"
    }
}
```

In Terminal it works, and I get:

```
$ /bin/composer.phar update -n -v
Loading composer repositories with package information
Updating dependencies (including require-dev)
  - Installing doctrine/common (2.3.0)
    Downloading: 100%         
    Extracting archive

  - Installing doctrine/mongodb (1.0.2)
    Downloading: 100%         
    Extracting archive

  - Installing symfony/console (v2.2.0)
    Loading from cache
    Extracting archive

  - Installing doctrine/mongodb-odm (dev-master ecb57e8)
    Cloning ecb57e843227269b3fed192dc6619f719a75dd0d
Cloning into '.../mongodb-odm'...

  - Marking doctrine/mongodb-odm (1.0.x-dev ecb57e8) as installed, alias of doctrine/mongodb-odm (dev-master ecb57e8)
doctrine/mongodb-odm suggests installing symfony/yaml (Enables the YAML metadata mapping driver)
Writing lock file
Generating autoload files
```

But in ST2 I get:

```
-- Starting Composer --
binary   : /bin/composer.phar
command  : update
arguments: -n -v

executing: /bin/composer.phar update -n -v
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - doctrine/mongodb-odm 1.0.0-BETA6 requires doctrine/mongodb >=1.0.0-beta1,<1.1-dev -> satisfiable by doctrine/mongodb 1.0.0, doctrine/mongodb 1.0.1, doctrine/mongodb 1.0.2.
    - doctrine/mongodb-odm 1.0.0-BETA7 requires doctrine/mongodb 1.0.* -> satisfiable by doctrine/mongodb 1.0.0, doctrine/mongodb 1.0.1, doctrine/mongodb 1.0.2.
    - doctrine/mongodb-odm 1.0.0-BETA8 requires doctrine/mongodb 1.0.* -> satisfiable by doctrine/mongodb 1.0.0, doctrine/mongodb 1.0.1, doctrine/mongodb 1.0.2.
    - doctrine/mongodb-odm 1.0.x-dev requires doctrine/mongodb 1.0.* -> satisfiable by doctrine/mongodb 1.0.0, doctrine/mongodb 1.0.1, doctrine/mongodb 1.0.2.
    - doctrine/mongodb 1.0.2 requires ext-mongo >=1.2.12,<1.4-dev -> the requested PHP extension mongo is missing from your system.
    - doctrine/mongodb 1.0.1 requires ext-mongo >=1.2.12,<1.4-dev -> the requested PHP extension mongo is missing from your system.
    - doctrine/mongodb 1.0.0 requires ext-mongo >=1.2.12,<1.4-dev -> the requested PHP extension mongo is missing from your system.
    - doctrine/mongodb-odm 1.0.0-BETA5 requires doctrine/mongodb 1.0.0-BETA1 -> no matching package found.
    - doctrine/mongodb-odm 1.0.0-BETA4 requires doctrine/mongodb 1.0.0-BETA1 -> no matching package found.
    - Installation request for doctrine/mongodb-odm 1.0.*@dev -> satisfiable by doctrine/mongodb-odm 1.0.0-BETA4, doctrine/mongodb-odm 1.0.0-BETA5, doctrine/mongodb-odm 1.0.0-BETA6, doctrine/mongodb-odm 1.0.0-BETA7, doctrine/mongodb-odm 1.0.0-BETA8, doctrine/mongodb-odm 1.0.x-dev.

Potential causes:
 - A typo in the package name
 - The package is not available in a stable-enough version according to your minimum-stability setting
   see <https://groups.google.com/d/topic/composer-dev/_g3ASeIFlrc/discussion> for more details.

Read <http://getcomposer.org/doc/articles/troubleshooting.md> for further common problems.
```

**Comment by francodacosta** [comment]
I tried it on my system and had no problems

```
-- Starting Composer --
binary   : /bin/composer.phar
command  : update
arguments: -n -v

executing: /bin/composer.phar update -n -v
Loading composer repositories with package information
Updating dependencies (including require-dev)
  - Installing doctrine/common (2.3.0)
    Loading from cache
    Extracting archive

  - Installing doctrine/mongodb (1.0.2)
    Loading from cache
    Extracting archive

  - Installing symfony/console (v2.2.0)
    Loading from cache
    Extracting archive

  - Installing doctrine/mongodb-odm (dev-master ecb57e8)
    Cloning ecb57e843227269b3fed192dc6619f719a75dd0d
Cloning into '/home/nuno/projects/personal/tests/vendor/doctrine/mongodb-odm'...

  - Marking doctrine/mongodb-odm (1.0.x-dev ecb57e8) as installed, alias of doctrine/mongodb-odm (dev-master ecb57e8)
doctrine/mongodb-odm suggests installing symfony/yaml (Enables the YAML metadata mapping driver)
Generating autoload files

-- Terminated -- 
```

I'm on Linux running sublime 2.0.1 build 2217 and Composer version d929a0813ae473272d151d9ebb2af7ebae451e48

I'm lost, I don't really know what is happening, it might be something specific with OSX, do have any ideas ?

nuno

**Comment by jacksleight** [comment]
I'll have a poke around and see if I can figure it out, I'm wondering if Composer is using a different PATH when running through ST2? OS X does have it's own PHP version, which I do not use (and doesn't have mongo enabled), however as far as I'm aware everything (including the default PATH) is set correctly and should be using the MacPorts version.

**Comment by francodacosta** [comment]
it might as well be that you are using a different php version in ST2.

can you try setting your composer path to a simple shell script 

```
{
    "composer_command":  "/bin/st_composer.sh"
}
```

ans st_composer.sh is something like

```
#!/bin/bash

/path/to/php /bin/composer.phar $@
```

**Comment by jacksleight** [comment]
Yup, can confirm that the default OS X php install is the problem (I added a new `show --platform` command to the plugin and mongo was missing). Instead of creating a script, I decided to remove the default php in `/usr/bin/php` and symlink `/usr/bin/php` to the MacPorts version at `/opt/local/bin/php`, so that it works for anything else that may be using the wrong binary.

Still be nice to figure out how to ensure `/opt/local/bin/php` is the only php version that's ever used, but that's obviously not an issue with your plugin.

Thanks for the help!
